src/main/java/seedu/address/logic/parser/FilterPetCommandParser.java

```diff
@@ -0,0 +1,75 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.util.function.Predicate;
+
+import seedu.address.logic.commands.FilterPetCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.pet.Pet;
+
+/**
+ * Parses input arguments and creates a new FilterPetCommand object.
+ */
+public class FilterPetCommandParser implements Parser<FilterPetCommand> {
+    public static final char COLOR_PREFIX = 'c';
+    public static final char PET_NAME_PREFIX = 'n';
+    public static final char PRICE_PREFIX = 'p';
+    public static final char SPECIES_PREFIX = 's';
+    public static final char VACCINATION_PREFIX = 'v';
```

Comment:
do we need a back slash for each prefix?

Reply:
idk I was thinking of something like find-p c/white, so in a command like 'find-p c/white s/cat', the args are split into 'c/white' and 's/cat', and the PredicateParser will take care of them respectively